src/zod-extensions.ts

```diff
@@ -181,4 +188,17 @@ export function extendZodWithOpenApi(zod: typeof z) {
 
     return result;
   };
+
+  zod.ZodUnion.prototype.openapiAs = function (
```

Comment:
Hmm, not 100% sure on including another function. Can this be done with the `.openapi` itself? Maybe as an additional parameter so that it doesn't clash with openapi fields?